crates/interfaces/src/p2p/full_block_range.rs

```diff
@@ -0,0 +1,237 @@
+use crate::p2p::{
+    bodies::client::BodiesClient, error::PeerRequestResult, headers::client::HeadersClient,
+};
+use reth_primitives::{BlockBody, Header, HeadersDirection, SealedBlock, SealedHeader, H256};
+use std::{
+    cmp::Reverse,
+    fmt::Debug,
+    future::Future,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+use tracing::debug;
+
+use super::headers::client::HeadersRequest;
+
+/// A Client that can fetch full blocks from the network.
+#[derive(Debug, Clone)]
+pub struct FullBlockRangeClient<Client> {
+    client: Client,
+}
+
+impl<Client> FullBlockRangeClient<Client> {
+    /// Creates a new instance of `FullBlockRangeClient`.
+    pub fn new(client: Client) -> Self {
+        Self { client }
+    }
+}
+
+impl<Client> FullBlockRangeClient<Client>
+where
+    Client: BodiesClient + HeadersClient + Clone,
+{
+    /// Returns a future that fetches [SealedBlock]s for the given hash and count.
+    ///
+    /// Note: this future is cancel safe
+    ///
+    /// Caution: This does no validation of body (transactions) responses but guarantees that
+    /// [SealedHeader]s match the requested hash.
+    pub fn get_full_block_range(
+        &self,
+        hash: H256,
+        count: u64,
+    ) -> FetchFullBlockRangeFuture<Client> {
```

Comment:
can even move this to the existing `FullBlockClient`?
then we don't need two client fields in the sync engine

Reply:
yeah that makes sense, will move